src/object/cli_obj.c

```diff
@@ -1724,6 +1725,8 @@ obj_req_fanout(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
 		}
 	}
 
+	D_ASSERT(!obj_auxi->args_initialized);
+
```

Comment:
this assertion probably should be removed, because it is possible that (obj_auxi->io_retry == 0 && obj_auxi->args_initialized). for example current dac_array_set_size implementation, it possibly will retry DAOS_OPC_OBJ_LIST_DKEY task by itself (see adjust_array_size_cb -> tse_task_reinit, that retry is not by obj client IO's automatically retry). 
I think this is why the Jenkins reports failure for this patch. I'll push another small commit to this PR to just remove this assertion.

Reply:
Thanks, I will update the patch to fix it.

Reply:
NP, I thought just now that probably can keep the assertion but zero the obj_auxi when need not retry, that maybe safer for user if it reuse/resched the task. But can be changed later in other patch to avoid update this patch again and slow the landing.

diff --git a/src/object/cli_obj.c b/src/object/cli_obj.c
index a3e44d0..3f78cd8 100644
--- a/src/object/cli_obj.c
+++ b/src/object/cli_obj.c
@@ -1725,6 +1725,8 @@ obj_req_fanout(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
                }
        }
 
+       D_ASSERT(!obj_auxi->args_initialized);
+
        /* if with only one target, need not to create separate shard task */
        if (tgts_nr == 1) {
                shard_auxi = obj_embedded_shard_arg(obj_auxi);
@@ -1978,6 +1980,10 @@ obj_comp_cb(tse_task_t *task, void *data)
                        D_ASSERT(d_list_empty(head));
                }
                obj_bulk_fini(obj_auxi);
+               /* zero it as your might reuse/resched the task, for example
+                * the usage in dac_array_set_size().
+                */
+               memset(obj_auxi, 0, sizeof(*obj_auxi));
        }
 
        obj_decref(obj);
@@ -2631,6 +2637,7 @@ dc_obj_query_key(tse_task_t *api_task)
                goto task_sched;
        }
 
+       D_ASSERT(!obj_auxi->args_initialized);
        D_ASSERT(d_list_empty(head));
 
        /* In each redundancy group, the QUERY RPC only needs to be sent